framework/platform/HLAPI/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+project(hlapi VERSION ${prplmesh_VERSION})
+
+message("${BoldWhite}Preparing ${BoldGreen}${PROJECT_NAME}${BoldWhite} for the ${BoldGreen}${TARGET_PLATFORM}${BoldWhite} platform${ColourReset}")
+
+# Set the base path for the current module
+set(MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
+
+# Common Sources
+file(GLOB_RECURSE hlapi_sources ${MODULE_PATH}/*.c*)
```

Comment:
Since even CMAKE developers are [recommending NOT to use GLOB](https://cmake.org/cmake/help/latest/command/file.html#filesystem) for picking up all the files to be compiled, and since I have some more reasons on top on their's, I think it might be a good opportunity to start explicitly add the files that are part of the library.
On the other hand, it will introduce inconsistency with other parts of the project that still use GLOB.
However, the reason to ignore the inconsistency, is that specifying the files explicitly does not require any special understanding, it is straightforward, no learning curve at all.

Reply:
Then let's start to get rid of GLOB. Suggestion applied. Resolving.